src/gp_util.c

```diff
@@ -7,9 +7,38 @@
 #include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
+#include <stdint.h>
+#include <time.h>
 
 #include "gp_common.h"
 
+#define USEC_INFINITY ((uint64_t)UINT64_MAX)
+#define NSEC_PER_USEC ((uint64_t)1000ULL)
+#define USEC_PER_SEC  ((uint64_t)1000000ULL)
+uint64_t time_now_usec(void)
+{
+    struct timespec ts;
+
+    if (clock_gettime(CLOCK_MONOTONIC, &ts) < 0) {
```

Comment:
does this really need to be the monotonic clock ?
Seem it is used only for logging, so perhaps the actual system time is a better identifier here?

Reply:
Yes, it's the protocol for `Type=notify-reload`, documented in `man systemd.service(5)` (on a sufficiently recent system). Basically I wrote it like this so that the systemd integration is backwards and forwards compatible (right now the patches only use `Type=notify` in the `.service` file for backwards compatibility, but the resulting code itself supports the newer `Type=notify-reload` protocol out-of-the-box)